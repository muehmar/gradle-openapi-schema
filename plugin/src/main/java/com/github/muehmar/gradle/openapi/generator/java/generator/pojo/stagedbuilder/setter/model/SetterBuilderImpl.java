package com.github.muehmar.gradle.openapi.generator.java.generator.pojo.stagedbuilder.setter.model;

import com.github.muehmar.gradle.openapi.generator.settings.PojoSettings;
import io.github.muehmar.codegenerator.writer.Writer;
import io.github.muehmar.pojobuilder.annotations.BuildMethod;
import io.github.muehmar.pojobuilder.annotations.PojoBuilder;
import java.util.function.Predicate;
import java.util.function.UnaryOperator;
import lombok.Value;

@PojoBuilder(builderName = "SetterBuilder")
@Value
public class SetterBuilderImpl {
  SetterType type;
  Predicate<SetterMember> includeInBuilder;
  String typeFormat;
  UnaryOperator<Writer> addRefs;

  public enum SetterType {
    DEFAULT,
    NULLABLE_ITEMS_LIST
  }

  @BuildMethod
  public static Setter buildSetter(SetterBuilderImpl setter) {
    return new Setter() {
      @Override
      public Setter forType(SetterType setterType) {
        return SetterBuilder.fullSetterBuilder()
            .type(setterType)
            .includeInBuilder(setter.includeInBuilder)
            .typeFormat(setter.typeFormat)
            .addRefs(setter.addRefs)
            .build();
      }

      @Override
      public boolean includeInBuilder(SetterMember member) {
        return setter.includeInBuilder.test(member)
            && (setter.type.equals(SetterType.DEFAULT)
                || member.getMember().getJavaType().isNullableItemsArrayType());
      }

      @Override
      public String methodName(SetterMember member, PojoSettings settings) {
        if (setter.type.equals(SetterType.DEFAULT)) {
          return member
              .getMember()
              .prefixedMethodName(settings.getBuilderMethodPrefix())
              .asString();
        } else {
          return member
              .getMember()
              .prefixedMethodName(settings.getBuilderMethodPrefix())
              .asString()
              .concat("_");
        }
      }

      @Override
      public String argumentType(SetterMember member) {
        final String unwrappedArgumentType;
        if (setter.type.equals(SetterType.DEFAULT)) {
          unwrappedArgumentType =
              member.getMember().getJavaType().getParameterizedClassName().asString();
        } else {
          unwrappedArgumentType =
              member
                  .getMember()
                  .getJavaType()
                  .getParameterizedClassName()
                  .asStringWrappingNullableValueType();
        }
        return String.format(setter.typeFormat, unwrappedArgumentType);
      }

      @Override
      public Writer addRefs(Writer writer) {
        return setter.addRefs.apply(writer);
      }
    };
  }
}
