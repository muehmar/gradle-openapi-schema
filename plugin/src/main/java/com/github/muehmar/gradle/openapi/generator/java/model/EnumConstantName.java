package com.github.muehmar.gradle.openapi.generator.java.model;

import com.github.muehmar.gradle.openapi.generator.java.model.name.JavaName;
import com.github.muehmar.gradle.openapi.generator.model.name.Name;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class EnumConstantName {
  private final String originalConstant;

  private EnumConstantName(String originalConstant) {
    this.originalConstant = originalConstant;
  }

  public static EnumConstantName ofString(String originalConstant) {
    return new EnumConstantName(originalConstant);
  }

  public String getOriginalConstant() {
    return originalConstant;
  }

  public JavaName asJavaConstant() {
    return JavaName.fromName(toUpperCaseSnakeCase(originalConstant));
  }

  private static Name toUpperCaseSnakeCase(String name) {
    if (name.toUpperCase().equals(name)) {
      return Name.ofString(name);
    }

    final String converted =
        name.trim()
            .replaceAll("([A-Z])", "_$1")
            .toUpperCase()
            .replaceFirst("^_", "")
            .replaceAll("_+", "_");
    return Name.ofString(converted);
  }
}
