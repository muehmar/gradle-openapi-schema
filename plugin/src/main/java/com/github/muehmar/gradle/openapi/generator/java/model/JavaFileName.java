package com.github.muehmar.gradle.openapi.generator.java.model;

import com.github.muehmar.gradle.openapi.generator.java.generator.parameter.JavaParameter;
import com.github.muehmar.gradle.openapi.generator.java.model.name.JavaName;
import com.github.muehmar.gradle.openapi.generator.java.model.pojo.JavaPojo;
import com.github.muehmar.gradle.openapi.generator.settings.PojoSettings;
import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class JavaFileName {
  private final Path path;

  private JavaFileName(Path path) {
    this.path = path;
  }

  public static JavaFileName fromRef(String ref) {
    return new JavaFileName(Paths.get(ref.replace(".", File.separator) + ".java"));
  }

  public static JavaFileName fromSettingsAndPojo(PojoSettings settings, JavaPojo pojo) {
    return fromSettingsAndClassname(settings, pojo.getClassName());
  }

  public static JavaFileName fromSettingsAndClassname(PojoSettings settings, JavaName className) {
    final String packagePath =
        settings
            .getPackageName()
            .replace(".", File.separator)
            .replaceFirst("^" + File.separator, "");
    final String fileNameString =
        String.format("%s%s%s.java", packagePath, File.separator, className);
    return new JavaFileName(Paths.get(fileNameString));
  }

  public static JavaFileName fromSettingsAndParameter(
      PojoSettings settings, JavaParameter parameter) {
    final String packagePath =
        settings
            .getPackageName()
            .replace(".", File.separator)
            .replaceFirst("^" + File.separator, "");
    final String fileNameString =
        String.format(
            "%s%sparameter%s%s.java",
            packagePath, File.separator, File.separator, parameter.getParamClassName());
    return new JavaFileName(Paths.get(fileNameString));
  }

  public Path asPath() {
    return path;
  }
}
