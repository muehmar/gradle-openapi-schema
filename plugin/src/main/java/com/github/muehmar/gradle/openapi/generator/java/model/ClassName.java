package com.github.muehmar.gradle.openapi.generator.java.model;

import ch.bluecare.commons.data.PList;
import com.github.muehmar.gradle.openapi.generator.model.Name;
import com.github.muehmar.gradle.openapi.generator.settings.ClassTypeMapping;
import com.github.muehmar.gradle.openapi.generator.settings.FormatTypeMapping;
import java.util.Optional;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class ClassName {
  private final Optional<PackageName> pkg;

  private final Name name;

  private ClassName(Optional<PackageName> pkg, Name name) {
    this.pkg = pkg;
    this.name = name;
  }

  private ClassName(Name name) {
    this(Optional.empty(), name);
  }

  private ClassName(PackageName pkg, Name name) {
    this(Optional.of(pkg), name);
  }

  public static ClassName ofQualifiedClassName(String qualifiedClassName) {
    final int i = qualifiedClassName.lastIndexOf(".");
    if (i > 0 && i < qualifiedClassName.length() - 1) {
      return new ClassName(
          PackageName.ofString(qualifiedClassName.substring(0, i)),
          Name.ofString(qualifiedClassName.substring(i + 1)));
    } else {
      return new ClassName(Name.ofString(qualifiedClassName));
    }
  }

  public static ClassName ofName(String name) {
    return new ClassName(Name.ofString(name));
  }

  public static ClassName ofName(Name name) {
    return new ClassName(name);
  }

  public static ClassName ofPackageAndName(PackageName pkg, Name name) {
    return new ClassName(pkg, name);
  }

  public static ClassName fromFormatTypeMapping(FormatTypeMapping formatTypeMapping) {
    return ClassName.ofQualifiedClassName(formatTypeMapping.getClassType());
  }

  public static ClassName fromClassTypeMapping(ClassTypeMapping classTypeMapping) {
    return ClassName.ofQualifiedClassName(classTypeMapping.getToClass());
  }

  public static Optional<ClassName> fromFormatTypeMapping(
      String formatString, PList<FormatTypeMapping> formatTypeMappings) {
    return formatTypeMappings
        .find(formatTypeMapping -> formatTypeMapping.getFormatType().equals(formatString))
        .map(ClassName::fromFormatTypeMapping);
  }

  public Name getClassName() {
    return name;
  }

  public Name getQualifiedClassName() {
    return pkg.map(p -> p.qualifiedClassName(name)).orElse(name);
  }

  public Name getClassNameWithGenerics(Name className, Name... more) {
    final PList<Name> generics = PList.fromArray(more).cons(className);
    final String genericsCommaSeparated = generics.map(Name::asString).mkString(", ");
    final String genericString =
        genericsCommaSeparated.isEmpty() ? "" : String.format("<%s>", genericsCommaSeparated);
    return getClassName().append(genericString);
  }

  public ClassName mapWithClassMappings(PList<ClassTypeMapping> classMappings) {
    return classMappings
        .filter(classMapping -> classMapping.getFromClass().equals(name.asString()))
        .headOption()
        .map(ClassName::fromClassTypeMapping)
        .orElse(this);
  }

  public ClassName asInnerClassOf(JavaIdentifier outerClassName) {
    return new ClassName(pkg, Name.ofString(outerClassName.asString()).append(".").append(name));
  }
}
