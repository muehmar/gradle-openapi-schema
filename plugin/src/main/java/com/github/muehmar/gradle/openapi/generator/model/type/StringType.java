package com.github.muehmar.gradle.openapi.generator.model.type;

import ch.bluecare.commons.data.PList;
import com.github.muehmar.gradle.openapi.generator.model.Nullability;
import com.github.muehmar.gradle.openapi.generator.model.Type;
import com.github.muehmar.gradle.openapi.generator.model.constraints.Constraints;
import com.github.muehmar.gradle.openapi.generator.model.name.PojoName;
import com.github.muehmar.gradle.openapi.generator.settings.PojoNameMapping;
import java.util.function.Function;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class StringType implements Type {
  private final Format format;
  private final String formatString;
  private final Nullability nullability;
  private final Constraints constraints;

  private StringType(
      Format format, String formatString, Nullability nullability, Constraints constraints) {
    this.format = format;
    this.formatString = formatString;
    this.nullability = nullability;
    this.constraints = constraints;
  }

  public static StringType noFormat() {
    return ofFormat(Format.NONE);
  }

  public static StringType uuid() {
    return ofFormat(Format.UUID);
  }

  public static StringType ofFormat(Format format) {
    return new StringType(format, format.value, Nullability.NOT_NULLABLE, Constraints.empty());
  }

  public static StringType ofFormatAndValue(Format format, String formatString) {
    return new StringType(format, formatString, Nullability.NOT_NULLABLE, Constraints.empty());
  }

  public Format getFormat() {
    return format;
  }

  public String getFormatString() {
    return formatString;
  }

  public StringType withNullability(Nullability nullability) {
    return new StringType(format, formatString, nullability, constraints);
  }

  public StringType withConstraints(Constraints constraints) {
    return new StringType(format, formatString, nullability, constraints);
  }

  public StringType addConstraints(Constraints constraints) {
    return new StringType(format, formatString, nullability, this.constraints.and(constraints));
  }

  @Override
  public Constraints getConstraints() {
    return constraints;
  }

  @Override
  public StringType applyMapping(PojoNameMapping pojoNameMapping) {
    return this;
  }

  @Override
  public Type makeNullable() {
    return withNullability(Nullability.NULLABLE);
  }

  @Override
  public Type replaceObjectType(
      PojoName objectTypeName, String newObjectTypeDescription, Type newObjectType) {
    return this;
  }

  @Override
  public Nullability getNullability() {
    return nullability;
  }

  @Override
  public <T> T fold(
      Function<NumericType, T> onNumericType,
      Function<IntegerType, T> onIntegerType,
      Function<StringType, T> onStringType,
      Function<ArrayType, T> onArrayType,
      Function<BooleanType, T> onBooleanType,
      Function<ObjectType, T> onObjectType,
      Function<EnumType, T> onEnumType,
      Function<MapType, T> onMapType,
      Function<AnyType, T> onAnyType,
      Function<MultiType, T> onMultiType) {
    return onStringType.apply(this);
  }

  public enum Format {
    DATE("date"),
    DATE_TIME("date-time"),
    TIME("partial-time"),
    PASSWORD("password"),
    BYTE("byte"),
    BINARY("binary"),
    EMAIL("email"),
    UUID("uuid"),
    URI("uri"),
    URL("url"),
    HOSTNAME("hostname"),
    IPV4("ipv4"),
    IPV6("ipv6"),
    NONE("none"),
    OTHER("other");

    private final String value;

    Format(String value) {
      this.value = value;
    }

    public String getValue() {
      return value;
    }

    public static Format parseString(String value) {
      return PList.fromArray(values()).find(f -> f.value.equals(value)).orElse(OTHER);
    }
  }
}
