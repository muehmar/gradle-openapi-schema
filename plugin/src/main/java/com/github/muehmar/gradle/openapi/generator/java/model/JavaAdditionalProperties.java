package com.github.muehmar.gradle.openapi.generator.java.model;

import static com.github.muehmar.gradle.openapi.generator.java.model.type.JavaAnyType.javaAnyType;
import static com.github.muehmar.gradle.openapi.util.Booleans.not;

import com.github.muehmar.gradle.openapi.generator.java.generator.enumpojo.EnumContentBuilder;
import com.github.muehmar.gradle.openapi.generator.java.generator.enumpojo.EnumGenerator;
import com.github.muehmar.gradle.openapi.generator.java.model.name.JavaName;
import com.github.muehmar.gradle.openapi.generator.java.model.type.JavaEnumType;
import com.github.muehmar.gradle.openapi.generator.java.model.type.JavaType;
import com.github.muehmar.gradle.openapi.generator.model.AdditionalProperties;
import com.github.muehmar.gradle.openapi.generator.settings.TypeMappings;
import java.util.Optional;
import java.util.function.Function;
import lombok.Value;

@Value
public class JavaAdditionalProperties {
  private static final JavaName MAP_PROPERTY_NAME = JavaName.fromString("additionalProperties");
  boolean allowed;
  JavaType type;

  public static JavaAdditionalProperties wrap(
      AdditionalProperties additionalProperties, TypeMappings typeMappings) {
    return new JavaAdditionalProperties(
        additionalProperties.isAllowed(),
        JavaType.wrap(additionalProperties.getType(), typeMappings));
  }

  public static JavaAdditionalProperties anyTypeAllowed() {
    return new JavaAdditionalProperties(true, javaAnyType());
  }

  public static JavaAdditionalProperties allowedFor(JavaType type) {
    return new JavaAdditionalProperties(true, type);
  }

  public static JavaAdditionalProperties notAllowed() {
    return new JavaAdditionalProperties(false, javaAnyType());
  }

  public static JavaName additionalPropertiesName() {
    return MAP_PROPERTY_NAME;
  }

  public boolean isNotAllowed() {
    return not(allowed);
  }

  public boolean isValueAnyType() {
    return type.equals(javaAnyType());
  }

  public boolean isNotValueAnyType() {
    return not(type.equals(javaAnyType()));
  }

  public TechnicalPojoMember asTechnicalPojoMember() {
    return TechnicalPojoMember.additionalProperties(type);
  }

  public Optional<EnumGenerator.EnumContent> asEnumContent() {
    final Function<JavaEnumType, Optional<EnumGenerator.EnumContent>> createEnumContent =
        enumType ->
            Optional.of(
                EnumContentBuilder.create()
                    .className(JavaName.fromName(type.getQualifiedClassName().getClassName()))
                    .description("Additional property enum")
                    .members(enumType.getMembers())
                    .build());
    return type.fold(
        ignore -> Optional.empty(),
        ignore -> Optional.empty(),
        createEnumContent,
        ignore -> Optional.empty(),
        ignore -> Optional.empty(),
        ignore -> Optional.empty(),
        ignore -> Optional.empty(),
        ignore -> Optional.empty(),
        ignore -> Optional.empty());
  }
}
