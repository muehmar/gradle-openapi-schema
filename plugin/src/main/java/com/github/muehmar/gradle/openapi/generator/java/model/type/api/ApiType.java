package com.github.muehmar.gradle.openapi.generator.java.model.type.api;

import ch.bluecare.commons.data.PList;
import com.github.muehmar.gradle.openapi.generator.java.model.name.ParameterizedApiClassName;
import com.github.muehmar.gradle.openapi.generator.java.model.name.QualifiedClassName;
import com.github.muehmar.gradle.openapi.generator.java.model.type.JavaType;
import com.github.muehmar.gradle.openapi.generator.settings.TypeConversion;
import lombok.Value;

@Value
public class ApiType {
  QualifiedClassName className;
  ParameterizedApiClassName parameterizedClassName;
  ToApiTypeConversion toApiTypeConversion;
  FromApiTypeConversion fromApiTypeConversion;

  public static ApiType fromConversion(
      QualifiedClassName className, TypeConversion conversion, PList<JavaType> generics) {
    final ParameterizedApiClassName parameterizedClassName =
        ParameterizedApiClassName.ofClassNameAndGenerics(className, generics);
    final ToApiTypeConversion toApiTypeConversion =
        new ToApiTypeConversion(ConversionMethod.ofString(className, conversion.getToCustomType()));
    final FromApiTypeConversion fromApiTypeConversion =
        new FromApiTypeConversion(
            ConversionMethod.ofString(className, conversion.getFromCustomType()));
    return new ApiType(
        className, parameterizedClassName, toApiTypeConversion, fromApiTypeConversion);
  }
}
