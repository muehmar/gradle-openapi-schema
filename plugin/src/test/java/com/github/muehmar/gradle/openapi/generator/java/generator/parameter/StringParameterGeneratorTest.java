package com.github.muehmar.gradle.openapi.generator.java.generator.parameter;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.github.muehmar.gradle.openapi.generator.model.Name;
import com.github.muehmar.gradle.openapi.generator.model.Parameter;
import com.github.muehmar.gradle.openapi.generator.model.constraints.Constraints;
import com.github.muehmar.gradle.openapi.generator.model.constraints.Pattern;
import com.github.muehmar.gradle.openapi.generator.model.constraints.Size;
import com.github.muehmar.gradle.openapi.generator.model.type.StringType;
import com.github.muehmar.gradle.openapi.generator.settings.TestPojoSettings;
import io.github.muehmar.codegenerator.writer.Writer;
import java.util.Optional;
import org.junit.jupiter.api.Test;

class StringParameterGeneratorTest {

  @Test
  void generate_when_noSize_then_correctRendered() {
    final ParameterGenerator gen = new ParameterGenerator();
    final Parameter param =
        new Parameter(Name.ofString("stringParam"), StringType.noFormat(), Optional.of("mode"));
    final JavaParameter limitParam = JavaParameter.wrap(param);

    final Writer writer =
        gen.generate(limitParam, TestPojoSettings.defaultSettings(), Writer.createDefault());

    assertEquals(
        "package com.github.muehmar.parameter;\n"
            + "\n"
            + "\n"
            + "public final class StringParam {\n"
            + "  private StringParam() {}\n"
            + "\n"
            + "  public static final String DEFAULT = \"mode\";\n"
            + "\n"
            + "  public static boolean matchesLimits(String val) {\n"
            + "    return true;\n"
            + "  }\n"
            + "\n"
            + "  public static boolean matchesPattern(String val) {\n"
            + "    return true;\n"
            + "  }\n"
            + "\n"
            + "}",
        writer.asString());
  }

  @Test
  void generate_when_minSize_then_correctRendered() {
    final ParameterGenerator gen = new ParameterGenerator();
    final Parameter param =
        new Parameter(
            Name.ofString("stringParam"),
            StringType.noFormat().withConstraints(Constraints.ofSize(Size.ofMin(1))),
            Optional.empty());
    final JavaParameter limitParam = JavaParameter.wrap(param);

    final Writer writer =
        gen.generate(limitParam, TestPojoSettings.defaultSettings(), Writer.createDefault());

    assertEquals(
        "package com.github.muehmar.parameter;\n"
            + "\n"
            + "\n"
            + "public final class StringParam {\n"
            + "  private StringParam() {}\n"
            + "\n"
            + "  public static final int MIN_LENGTH = 1;\n"
            + "\n"
            + "  public static boolean matchesLimits(String val) {\n"
            + "    return MIN_LENGTH <= val.length();\n"
            + "  }\n"
            + "\n"
            + "  public static boolean matchesPattern(String val) {\n"
            + "    return true;\n"
            + "  }\n"
            + "\n"
            + "}",
        writer.asString());
  }

  @Test
  void generate_when_maxSize_then_correctRendered() {
    final ParameterGenerator gen = new ParameterGenerator();
    final Parameter param =
        new Parameter(
            Name.ofString("stringParam"),
            StringType.noFormat().withConstraints(Constraints.ofSize(Size.ofMax(20))),
            Optional.empty());
    final JavaParameter limitParam = JavaParameter.wrap(param);

    final Writer writer =
        gen.generate(limitParam, TestPojoSettings.defaultSettings(), Writer.createDefault());

    assertEquals(
        "package com.github.muehmar.parameter;\n"
            + "\n"
            + "\n"
            + "public final class StringParam {\n"
            + "  private StringParam() {}\n"
            + "\n"
            + "  public static final int MAX_LENGTH = 20;\n"
            + "\n"
            + "  public static boolean matchesLimits(String val) {\n"
            + "    return val.length() <= MAX_LENGTH;\n"
            + "  }\n"
            + "\n"
            + "  public static boolean matchesPattern(String val) {\n"
            + "    return true;\n"
            + "  }\n"
            + "\n"
            + "}",
        writer.asString());
  }

  @Test
  void generate_when_minAndMaxAndDefault_then_correctRendered() {
    final ParameterGenerator gen = new ParameterGenerator();
    final Parameter param =
        new Parameter(
            Name.ofString("stringParam"),
            StringType.noFormat().withConstraints(Constraints.ofSize(Size.of(1, 20))),
            Optional.of("mode"));
    final JavaParameter limitParam = JavaParameter.wrap(param);

    final Writer writer =
        gen.generate(limitParam, TestPojoSettings.defaultSettings(), Writer.createDefault());

    assertEquals(
        "package com.github.muehmar.parameter;\n"
            + "\n"
            + "\n"
            + "public final class StringParam {\n"
            + "  private StringParam() {}\n"
            + "\n"
            + "  public static final int MIN_LENGTH = 1;\n"
            + "  public static final int MAX_LENGTH = 20;\n"
            + "  public static final String DEFAULT = \"mode\";\n"
            + "\n"
            + "  public static boolean matchesLimits(String val) {\n"
            + "    return MIN_LENGTH <= val.length() && val.length() <= MAX_LENGTH;\n"
            + "  }\n"
            + "\n"
            + "  public static boolean matchesPattern(String val) {\n"
            + "    return true;\n"
            + "  }\n"
            + "\n"
            + "}",
        writer.asString());
  }

  @Test
  void generate_when_pattern_then_correctRendered() {
    final ParameterGenerator gen = new ParameterGenerator();
    final Parameter param =
        new Parameter(
            Name.ofString("stringParam"),
            StringType.noFormat()
                .withConstraints(Constraints.ofPattern(Pattern.ofUnescapedString("[A-Za-z]\\d"))),
            Optional.empty());
    final JavaParameter limitParam = JavaParameter.wrap(param);

    final Writer writer =
        gen.generate(limitParam, TestPojoSettings.defaultSettings(), Writer.createDefault());

    assertEquals(
        "package com.github.muehmar.parameter;\n"
            + "\n"
            + "import java.util.regex.Pattern;\n"
            + "\n"
            + "public final class StringParam {\n"
            + "  private StringParam() {}\n"
            + "\n"
            + "  public static final Pattern PATTERN = Pattern.compile(\"[A-Za-z]\\\\d\");\n"
            + "  public static final String PATTERN_STR = \"[A-Za-z]\\\\d\";\n"
            + "\n"
            + "  public static boolean matchesLimits(String val) {\n"
            + "    return true;\n"
            + "  }\n"
            + "\n"
            + "  public static boolean matchesPattern(String val) {\n"
            + "    return PATTERN.matcher(val).matches();\n"
            + "  }\n"
            + "\n"
            + "}",
        writer.asString());
  }

  @Test
  void generate_when_dateFormat_then_correctRendered() {
    final ParameterGenerator gen = new ParameterGenerator();
    final Parameter param =
        new Parameter(
            Name.ofString("dateParam"),
            StringType.ofFormat(StringType.Format.DATE),
            Optional.empty());
    final JavaParameter limitParam = JavaParameter.wrap(param);

    final Writer writer =
        gen.generate(limitParam, TestPojoSettings.defaultSettings(), Writer.createDefault());

    assertEquals(
        "package com.github.muehmar.parameter;\n"
            + "\n"
            + "\n"
            + "public final class DateParam {\n"
            + "  private DateParam() {}\n"
            + "\n"
            + "\n"
            + "}",
        writer.asString());
  }
}
