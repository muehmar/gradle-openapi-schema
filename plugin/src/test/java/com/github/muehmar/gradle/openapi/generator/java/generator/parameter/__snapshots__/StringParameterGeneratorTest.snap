allStringTypeFormats[binary]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
}
]


allStringTypeFormats[byte]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
  public static boolean matchesLimits(String val) {
    return true;
  }

  public static boolean matchesPattern(String val) {
    return true;
  }

}
]


allStringTypeFormats[date-time]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
}
]


allStringTypeFormats[date]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
}
]


allStringTypeFormats[email]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
  public static boolean matchesLimits(String val) {
    return true;
  }

  public static boolean matchesPattern(String val) {
    return true;
  }

}
]


allStringTypeFormats[hostname]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
  public static boolean matchesLimits(String val) {
    return true;
  }

  public static boolean matchesPattern(String val) {
    return true;
  }

}
]


allStringTypeFormats[ipv4]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
  public static boolean matchesLimits(String val) {
    return true;
  }

  public static boolean matchesPattern(String val) {
    return true;
  }

}
]


allStringTypeFormats[ipv6]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
  public static boolean matchesLimits(String val) {
    return true;
  }

  public static boolean matchesPattern(String val) {
    return true;
  }

}
]


allStringTypeFormats[none]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
  public static boolean matchesLimits(String val) {
    return true;
  }

  public static boolean matchesPattern(String val) {
    return true;
  }

}
]


allStringTypeFormats[other]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
  public static boolean matchesLimits(String val) {
    return true;
  }

  public static boolean matchesPattern(String val) {
    return true;
  }

}
]


allStringTypeFormats[partial-time]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
}
]


allStringTypeFormats[password]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
  public static boolean matchesLimits(String val) {
    return true;
  }

  public static boolean matchesPattern(String val) {
    return true;
  }

}
]


allStringTypeFormats[uri]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
}
]


allStringTypeFormats[url]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
}
]


allStringTypeFormats[uuid]=[
package com.github.muehmar.parameter;

/**
 * @deprecated The generation of parameters will be removed with the next major
 * version 4.x of the plugin. Consider using the official openapi generator plugin
 * to generate the endpoints from the openapi specification while using the DTO's
 * generated by this plugin. You can find an example with Spring Boot in the GitHub
 * repository of this plugin.
 */
@Deprecated
public final class StringParam {
  private StringParam() {}
.
.
}
]